### PR TITLE
Various ZPL improvements and bug fixes

### PR DESCRIPTION
**Pull request by fkryvyts-ingrid**

### CODE REVIEW

File: drawers/barcode_aztec.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+			if !ok {
+				return nil
+			}
+
+			layers := 0
+
+			if barcode.Size > 0 {
+				switch {
+				case barcode.Size >= 200 && barcode.Size <= 232:
+					layers = barcode.Size - 200
```
Comment (mknycha-ingrid) [inline drawers/barcode_aztec.go:25]: nitpick: move 200 to a constant

File: hex/decode.go [hunk 2]
```diff
@@ -59,3 +59,101 @@ func DecodeEmbeddedImage(value string) ([]byte, error) {
...
+
+		if char == '!' {
+			if rowHex > line.Len() {
+				line.WriteString(strings.Repeat("1", rowHex-line.Len()))
+			}
+
+			continue
+		}
+
+		if char == ':' {
```
Comment (mknycha-ingrid) [inline hex/decode.go:145]: use `switch case`

File: parsers/barcode_aztec.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			}
+
+			if len(parts) > 1 {
+				if v, err := strconv.Atoi(parts[1]); err == nil {
+					barcode.Magnification = v
+				}
+			}
+
+			if len(parts) > 3 {
+				if v, err := strconv.Atoi(parts[3]); err == nil {
```
Comment (mknycha-ingrid) [inline parsers/barcode_aztec.go:32]: what is under `parts[2]` and why it's skipped here?

Comment (fkryvyts-ingrid) [reply parsers/barcode_aztec.go:32]: It is just some option that we do not support
```
eci: Whether or not the bar code data uses ECI (extended channel interpretation) to switch character sets. Valid values are Y and N. The default value is N (no ECI).
```

File: drawers/barcode_aztec.go (hunk 1 above)
Comment (fkryvyts-ingrid) [reply drawers/barcode_aztec.go]: moved

File: hex/decode.go (hunk 2 above)
Comment (fkryvyts-ingrid) [reply hex/decode.go]: used

File: parsers/barcode_aztec.go (hunk 3 above)
Comment (fkryvyts-ingrid) [reply parsers/barcode_aztec.go:34]: added TODO for handling eci

**Review by mknycha-ingrid — APPROVED** [review]

**Pull request merged**